arch/arm/core/cortex_m/Kconfig

```diff
@@ -129,15 +129,6 @@ config STACK_ALIGN_DOUBLE_WORD
 	This is needed to conform to AAPCS, the procedure call standard for
 	the ARM. It wastes stack space.
 
-config NUM_IRQ_PRIO_BITS
```

Comment:
Not sure why it didn't depend on !HAS_DTS.  The other few places I'll cleanup with more specific SoC related patches.